src/modules/esl-share/test/buttons/button-test-plan-factory.ts

```diff
@@ -0,0 +1,50 @@
+import {ESLShareConfig} from '../../core/esl-share-config';
+import {ESLShareActionRegistry} from '../../core/esl-share-action-registry';
+
+export function createImportCheckTestPlan(actionName: string): () => void {
+  return () => {
+    test('config instance has 1 button', () => {
+      expect(ESLShareConfig.instance.buttons.length).toBe(1);
+    });
+
+    test(`"${actionName}" action was registered when importing the button`, () => {
+      expect(ESLShareActionRegistry.instance.has(actionName)).toBe(true);
+    });
+  };
+}
+
+export function createButtonMatchingTestPlan(name: string, action: string): () => void {
+  return () => {
```

Comment:
Cosmetic and can be moved to the separate update:
Jest has built in tools to declare shape validation rules
```ts
interface SomeObjType {
    a: number;
    b: string;
    subObj: {
      c: number;
    };
    tuple: [string, string];
    optional?: string;
  }

  const SomeObjTypeShape: Readonly<SomeObjType> = expect.objectContaining({
    a: expect.any(Number),
    b: expect.any(String),
    subObj: expect.objectContaining({
      c: expect.any(Number),
    }),
    tuple: expect.arrayContaining([expect.any(String), expect.any(String)])
    // Skip optional propertys
  });

  test('Shape match sample', async () => {
    const objLikeShape = {
      a: 1,
      b: '2',
      subObj: {
        c: 3
      },
      tuple: ['4', '5']
    };

    expect(objLikeShape).toEqual(SomeObjTypeShape);
    expect({a: 1}).not.toEqual(SomeObjTypeShape);
  });
```

so all non-dunamic (non parametrized) tests may look like
```ts
   test(
      `The button for ${name} accepts ESLShareButtonConfig shape`, 
      () => expect(ESLShareConfig.instance.getButton(name)).toEqual(ESLShareButtonConfigShape);
   );
```

Reply:
Thanks a lot! I will look at it in deep in the next waves.